sender_pool.go

```diff
@@ -146,35 +171,49 @@ func (p *LineSenderPool) Acquire(ctx context.Context) (LineSender, error) {
 				return nil, errors.New("tcp/s not supported for pooled senders, use http/s only")
 			}
 		}
-		return newHttpLineSender(conf)
+		s, err = newHttpLineSender(conf)
+	}
+
+	if err != nil {
+		return nil, err
 	}
 
+	p.numSenders++
+
+	ps := &pooledSender{
+		pool:    p,
+		wrapped: s,
+	}
+	atomic.AddUint64(&ps.tick, 1)
+	return ps, nil
 }
 
-// Release flushes the LineSender and returns it back to the pool. If the pool
-// is full, the sender is closed and discarded. In cases where the sender's
-// flush fails, it is not added back to the pool.
-func (p *LineSenderPool) Release(ctx context.Context, s LineSender) error {
-	// If there is an error on flush, do not add the sender back to the pool
-	if err := s.Flush(ctx); err != nil {
-		return err
+func (p *LineSenderPool) free(ctx context.Context, ps *pooledSender) error {
+	var flushErr error
+
+	if ps.dirty {
+		flushErr = ps.Flush(ctx)
```

Comment:
Should we prevent dirty senders from flushing in `Flush()`? This way, we don't flush, receive an error, and then attempt to flush again?

Or is this just a best-effort thing?

Reply:
Done in 5866dcebba9dd4e7c5061e689ad97ca1754cb909